pages/submission/PKPSubmissionHandler.php

```diff
@@ -10,257 +10,838 @@
+
+    /**
+     * Get the state needed for the SubmissionFilesListPanel component
+     */
+    protected function getSubmissionFilesListPanel(Request $request, Submission $submission, array $genres): array
+    {
+        $submissionFiles = Repo::submissionFile()
+            ->getCollector()
+            ->filterBySubmissionIds([$submission->getId()])
+            ->filterByFileStages([SubmissionFile::SUBMISSION_FILE_SUBMISSION])
+            ->getMany();
+
+        // Don't allow dependent files to be uploaded with the submission
+        $genres = array_values(
+            array_filter($genres, fn ($genre) => !$genre->getDependent())
+        );
+
+        $form = new PKPSubmissionFileForm(
+            $this->getSubmissionFilesApiUrl($request, $submission->getId()),
+            $genres
+        );
+
+        return [
+            'addFileLabel' => __('common.addFile'),
+            'apiUrl' => $this->getSubmissionFilesApiUrl($request, $submission->getId()),
+            'cancelUploadLabel' => __('form.dropzone.dictCancelUpload'),
+            'genrePromptLabel' => __('submission.submit.genre.label'),
+            'documentTypes' => [
+                'DOCUMENT_TYPE_DEFAULT' => FileManager::DOCUMENT_TYPE_DEFAULT,
+                'DOCUMENT_TYPE_AUDIO' => FileManager::DOCUMENT_TYPE_AUDIO,
+                'DOCUMENT_TYPE_EXCEL' => FileManager::DOCUMENT_TYPE_EXCEL,
+                'DOCUMENT_TYPE_HTML' => FileManager::DOCUMENT_TYPE_HTML,
+                'DOCUMENT_TYPE_IMAGE' => FileManager::DOCUMENT_TYPE_IMAGE,
+                'DOCUMENT_TYPE_PDF' => FileManager::DOCUMENT_TYPE_PDF,
+                'DOCUMENT_TYPE_WORD' => FileManager::DOCUMENT_TYPE_WORD,
+                'DOCUMENT_TYPE_EPUB' => FileManager::DOCUMENT_TYPE_EPUB,
+                'DOCUMENT_TYPE_VIDEO' => FileManager::DOCUMENT_TYPE_VIDEO,
+                'DOCUMENT_TYPE_ZIP' => FileManager::DOCUMENT_TYPE_ZIP,
+            ],
+            'emptyLabel' => __('submission.upload.instructions'),
+            'emptyAddLabel' => __('common.upload.addFile'),
+            'fileStage' => SubmissionFile::SUBMISSION_FILE_SUBMISSION,
+            'form' => $form->getConfig(),
+            'genres' => array_map(
+                fn ($genre) => [
+                    'id' => (int) $genre->getId(),
+                    'name' => $genre->getLocalizedName(),
+                    'isPrimary' => !$genre->getSupplementary() && !$genre->getDependent(),
+                ],
+                $genres
+            ),
+            'id' => 'submissionFiles',
+            'items' => Repo::submissionFile()
+                ->getSchemaMap()
+                ->summarizeMany($submissionFiles, $genres)
+                ->values(),
+            'options' => [
+                'maxFilesize' => Application::getIntMaxFileMBs(),
+                'timeout' => ini_get('max_execution_time') ? ini_get('max_execution_time') * 1000 : 0,
+                'dropzoneDictDefaultMessage' => __('form.dropzone.dictDefaultMessage'),
```

Comment:
Yes, this is on my list of "hard problems" for the UI Library at the moment. I have gone back and forth and so there are two approaches mixed in right now:

1. Common locale keys are added to state through the TemplateManager and can be used in Vue components with `this.__('common.close')`. See https://github.com/pkp/pkp-lib/blob/main/classes/template/PKPTemplateManager.php#L814.
2. Locale keys used by a single component are passed in as props, like `dropzoneDictDefaultMessage`. The benefit of this approach is that we can track every use of a locale key. Although the prop name is different, it must be tied to a server-side use of the locale key. That means that we can track all uses within the PHP code. With the global locale keys set in `TemplateManager`, we have to search the UI Library and template files to see if they are used. The other downside of using global locale keys is that UI components become more tightly coupled to a specific translation. Often we want a component to be reusable in different contexts, which may require different labels.

I would love to be able to use an approach like https://github.com/avast/vuei18n-po, but I've always thought that we have just too many locale strings. All .po files except `emails.po` in locale/en_US and lib/pkp/locale/en_US add up to about 530kb, and I don't think a minified JSON file would save more than 10-20%. That's a pretty big increase in our payload, which is already far too big (builds at 2.5mb now). We need to start reducing the built file size.

For loading locale files, none of the approaches I can think of sound good: a) segment our .po files so that what's needed in state can be bundled with the JS package, but that seems hard and prone to errors; b) request locale keys "on demand" from the Vue components, but that would require quick server response times which aren't common in our community; c) pass locale keys into components directly, like we do now, but in a way that preserves the locale key structure (eg - a global state object of some kind, which is kind of how `$templateMgr->setLocaleKeys(...)` works, but figure out a better way to do that).

Reply:
Another thing we can do, which would be difficult but I think would be possible, is to write an extension to our webpack build script which searched our component files for a syntax (like `this.__(...)`), extracted the locale keys, and generated built files of just the locale keys that are used. We'd also need to scan `.tpl` files. And it wouldn't cover plugins. And I have no idea how to do it. But I'm pretty sure it's possible. :laughing:

Reply:
Let's leave that as a dev-infrastructure problem for later. I would be happy to start by prioritizing a locale file cleanup, to see what we're left with after that.